### PR TITLE
validate: add weightDevice validation

### PR DESCRIPTION
**Pull request by q384566678**
According to the following [specifications:](https://github.com/opencontainers/runtime-spec/blame/master/config-linux.md#L338)
``
You MUST specify at least one of weight or leafWeight in a given entry, and MAY specify both.
``

implement BlkIOWeightOrLeafWeightExist.

Signed-off-by: Zhou Hao <zhouhao@cn.fujitsu.com>

### CODE REVIEW

File: validate/validate_test.go [hunk 1]
```diff
@@ -557,6 +557,26 @@ func TestCheckLinux(t *testing.T) {
...
 		},
+		{
+			val: rspec.Spec{
+				Version: "1.0.0",
+				Linux: &rspec.Linux{
+					Resources: &rspec.LinuxResources{
+						BlockIO: &rspec.LinuxBlockIO{
+							WeightDevice: []rspec.LinuxWeightDevice{
+								{
+									{
```
Comment (wking) [inline validate/validate_test.go]: From [Travis][1]:

```
validate/validate_test.go:568:10: missing type in composite literal
validate/validate_test.go:568:10: implicit assignment of unexported field 'linuxBlockIODevice' in specs.LinuxWeightDevice literal
```

[1]: https://travis-ci.org/opencontainers/runtime-tools/jobs/337923884#L556

File: validate/validate.go [hunk 2]
```diff
@@ -817,6 +817,18 @@ func (v *Validator) CheckLinuxResources() (errs error) {
 		}
 	}
 
+	if r.BlockIO != nil && r.BlockIO.WeightDevice != nil {
+		for _, weightDevice := range r.BlockIO.WeightDevice {
+			if weightDevice.Weight == nil && weightDevice.LeafWeight == nil {
+				errs = multierror.Append(errs,
+					specerror.NewError(
+						specerror.BlkIOWeightOrLeafWeightExist,
+						fmt.Errorf("You MUST specify at least one of `weight` or `leafWeight` in a given entry"),
```
Comment (wking) [inline validate/validate.go]: This string is already covered by `BlkIOWeightOrLeafWeightExist`.  I think we want something like:

```go
for i, weightDevice := range r.BlockIO.WeightDevice {
    …
        fmt.Errorf("linux.resources.blockIO.weightDevice[%d] specifies neither weight nor leafWeight", i)
    …
}

Comment (q384566678) [reply validate/validate.go]: updated, thanks.

File: validate/validate_test.go (hunk 1 above)
Comment (q384566678) [reply validate/validate_test.go]: I don't know what to do with this, but I tried the following calls, and they all went wrong.
```
 WeightDevice: []rspec.LinuxWeightDevice{
        linuxBlockIODevice{
                Major: 5,
                Minor: 0,
        },
}
```
```
 WeightDevice: []rspec.LinuxWeightDevice{
        rspec.linuxBlockIODevice{
                Major: 5,
                Minor: 0,
        },
}
```

Comment (wking) [reply validate/validate_test.go]: > I don't know what to do with this, but I tried the following calls, and they all went wrong.

I'm pretty sure you want:

```go
WeightDevice: []rspec.LinuxWeightDevice{
  rspec.LinuxWeightDevice{
    Major: 5,
    Minor: 0,
  },
},
```

Comment (q384566678) [reply validate/validate_test.go]: This is still wrong.

Comment (wking) [reply validate/validate_test.go]: > This is still wrong.

Hmm.  Maybe Go has a problem with literal struct initialization for lowercase embedded types.  We may need to skip these for the literal initialization and add them later via the promoted properties:

```go
weightDevices := []rspec.LinuxWeightDevice{
  rspec.LinuxWeightDevice{},
}
weightDevices[0].Major = 5
weightDevices[0].Minor = 0
```

Then use `WeightDevice: weightDevices` when you initialize your config struct.

Comment (q384566678) [reply validate/validate_test.go]: Cool. It worked, thanks.

**Comment by q384566678** [comment]
@liangchenye PTAL

**Comment by q384566678** [comment]
ping @liangchenye

**Comment by liangchenye** [comment]
I'm working on implement 'kill' this morning.
I'll review it this afternoon. @q384566678

**Comment by liangchenye** [comment]
LGTM

**Pull request merged**